jobqueue_features/clusters.py

```diff
@@ -687,4 +727,54 @@ def _update_script_nodes(self, **kwargs) -> None:
             self._kwargs["command_template"] = command_template
 
 
-ClusterType = TypeVar("ClusterType", JobQueueCluster, LocalCluster, CustomSLURMCluster)
+class CustomPBSCluster(CustomClusterMixin, PBSCluster):
+    __doc__ = f"""Custom PBS Cluster class.
+
+    Attributes:
+    ---------
+    ngpus_per_node : int
+        The number of gpus per node to be used
+    {custom_cluster_attributes}
+    """
+
+    job_cls = CustomPBSJob
+
+    def __init__(self, **kwargs):
+        self.scheduler_name = "pbs"
+        self.ngpus_per_node = kwargs.get("ngpus_per_node", 0)
+        kwargs = self.update_init_kwargs(**kwargs)
+        self.validate_cluster_name(kwargs["name"])
+        self.name = kwargs["name"]
+        if self.mpi_mode:
+            if "job_cpu" in kwargs:
+                raise ValueError(
+                    "We don't allow 'job_cpu' as a kwarg in MPI mode since we"
+                    " leverage this in jobqueue to set cpus-per-task"
+                )
+            mpi_job_extra = []
+            if "job_extra" not in kwargs:
+                raise KeyError("job_extra keyword should always be set in kwargs")
+            mpi_job_extra.extend(kwargs["job_extra"])
+            kwargs.update({"job_extra": mpi_job_extra})
+        super().__init__(**kwargs)
+        if hasattr(self, "mpi_tasks"):
```

Comment:
This should probably be protected since it is only relevant in `mpi_mode`

Reply:
Good point. I will add that.

Reply:
Changed in https://github.com/E-CAM/jobqueue_features/pull/52/commits/337d29f8c02c9eeeb926c293f1b2fcc22f822105